python/semantic_kernel/connectors/openai_plugin/openai_function_execution_parameters.py

```diff
@@ -2,16 +2,47 @@
 
 
 from collections.abc import Awaitable, Callable
-from typing import Any
+from typing import TYPE_CHECKING, Any
+from urllib.parse import urlparse
 
-from semantic_kernel.connectors.openapi_plugin.openapi_function_execution_parameters import (
-    OpenAPIFunctionExecutionParameters,
-)
+import httpx
+from pydantic import Field
+from typing_extensions import deprecated
+
+from semantic_kernel.kernel_pydantic import KernelBaseModel
+
+if TYPE_CHECKING:
+    from semantic_kernel.connectors.openapi_plugin import (
+        OperationSelectionPredicateContext,
+    )
 
 OpenAIAuthCallbackType = Callable[..., Awaitable[Any]]
 
 
-class OpenAIFunctionExecutionParameters(OpenAPIFunctionExecutionParameters):
+@deprecated(
```

Comment:
Just curious: why are we adding a bunch of properties to this class while deprecating it?

Reply:
There was a circular dependency that I wanted to break, so I decided to remove the inheritance from OpenAPI settings. We will eventually get rid of this class in a version or two, so it should be short lived.